nilmtk/api.py

```diff
@@ -244,106 +243,60 @@ def train_jointly(self,clf,d):
 
     
     def test_jointly(self,d):
-        if self.site_only == True:
-            for dataset in d:
-                print("Loading data for ",dataset, " dataset")
-                test=DataSet(d[dataset]['path'])
+        # store the test_main readings for all buildings
+        for dataset in d:
+            print("Loading data for ",dataset, " dataset")
+            test=DataSet(d[dataset]['path'])
             for building in d[dataset]['buildings']:
                 test.set_window(start=d[dataset]['buildings'][building]['start_time'],end=d[dataset]['buildings'][building]['end_time'])
                 test_mains=next(test.buildings[building].elec.mains().load(physical_quantity='power', ac_type='apparent', sample_period=self.sample_period))
-                # print(f'test_mains: {test_mains}')
-
-                
                 if self.DROP_ALL_NANS:
-                    test_mains= self.dropna(test_mains)
+                    test_mains, _= self.dropna(test_mains,[])
```

Comment:
In the case `site_only == False`, `self.dropna()` is called twice. You can improve the code efficiency by changing this `if` condition in:
```python
if self.DROP_ALL_NANS and self.site_only:
```

Reply:
Done as u said :)